File: Arrays/CPP/Pascal'sTriangle.cpp
/*
Leetcode Question:
https://leetcode.com/problems/pascals-triangle/
*/
#include <vector>
using namespace std;

class Solution {
public:
    static vector<int> listElements(int row) {
        vector<int> answer;
        long long temp = 1;
        answer.push_back(1);
        for (int col = 1; col < row; col++) {
            temp = temp * (row - col);
            temp /= col;
            answer.push_back(static_cast<int>(temp));
        }
        return answer;
    }

    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        for (int i = 1; i <= numRows; i++) {
            ans.push_back(listElements(i));
        }
        return ans;
    }
};

File: Arrays/Java/Pascal'sTriangle.java
/*
Leetcode Question :
leetcode.com/problems/pascals-triangle/
*/
class Solution {
    public static List<Integer> listElements(int row) {
        ArrayList<Integer> answer = new ArrayList<>();
        long temp = 1;
        answer.add(1);
        for(int col = 1; col < row; col++) {
            temp = temp * (row - col);
            temp /= col;
            answer.add((int) temp);
        }
        return answer;
    }
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        for(int i = 1; i <= numRows; i++) {
            ans.add(listElements(i));
        }
        return ans;
    }
}
